webknossos-datastore/app/com/scalableminds/webknossos/datastore/datavault/VaultPath.scala

```diff
@@ -0,0 +1,76 @@
+package com.scalableminds.webknossos.datastore.datavault
+
+import com.scalableminds.util.io.ZipIO
+import com.scalableminds.webknossos.datastore.storage.FileSystemCredential
+import net.liftweb.util.Helpers.tryo
+
+import java.net.URI
+import java.nio.file.{FileSystem, LinkOption, Path, Paths, WatchEvent, WatchKey, WatchService}
+
+/*
+VaultPath implements Path so that a drop in replacement is possible while continuing to use Paths for local storage.
+This class does not implement all relevant methods and it might be a good idea to remove the inheritance on Path in the
+future.
+ */
+
+class VaultPath(uri: URI, dataVault: DataVault, fileSystemCredentialOpt: Option[FileSystemCredential]) extends Path {
```

Comment:
Isn’t the fileSystemCredentialOpt already stored in the DataVault? If so, I’d not store it again here, but rather provide an accessor if needed, that forwards this to dataVault.fileSystemCredentialOpt